R/fit.R

```diff
@@ -219,9 +227,18 @@ mmrm <- function(formula,
 
   attr(data, which = "dataname") <- toString(match.call()$data)
 
+  nobs <- nrow(data)
+  if (is.null(weights)){
+    weights <- rep(1, nobs)
+  }
+  if(!is.null(weights) && any(weights <= 0)){
+    stop("only positive weights are allowed")
+  }
```

Comment:
I would assert this in lower level function e.g. `h_mmrm_tmb()`, so not needed here
```suggestion
```